providers/utils.go

```diff
@@ -387,6 +397,31 @@ func createOAuth2ClientCredentialsHTTPClient( //nolint:ireturn
 	return oauthClient, nil
 }
 
+func createOAuth2PasswordHTTPClient(ctx context.Context, client *http.Client,
+	dbg bool, params *Oauth2PasswordParams,
+) (common.AuthenticatedHTTPClient, error) {
+	options := []common.OAuthOption{
+		common.WithOAuthClient(getClient(client)),
+		common.WithTokenSourceProvider(func() (*oauth2.Token, error) {
+			// we could use refresh token, but it works a bit differently when refresh token was
+			// acquired via oauth2 password grant type.
+			// TODO this is called every time a request is made. Store AT/RT
```

Comment:
what's AT/RT? Can you update the comment to make it more clear?

Reply:
sorry I use AT/RT a short version of access token and refresh token